libs/pgwire/src/main/antlr/PgArray.g4

```diff
@@ -22,40 +22,41 @@ array
    ;
 
 item
-   : NULL           #null
-   | string         #value
+   : string         #value
    | NUMBER         #value
-   | bool           #value
    | array          #noop
+   | 'true'         #value
+   | 'false'        #value
+   | 'NULL'         #null
    ;
 
-bool
-    : 'true'
-    | 'false'
-    ;
-
 
 string
     : STRING        #quotedString
-    | CHAR+         #unquotedString
+    | QSTRING       #unquotedString
     ;
 
 STRING
     : '"' (ESC | ~["\\])* '"'
     ;
 
-CHAR
-    : [!-z]
+
+QSTRING
+    : DIGIT* CHAR+
     ;
 
+CHAR
+    : [a-zA-Z]
+    | [!-+]
+    | '-'
+    | '_'
```

Comment:
Not sure if this is too restrictive, but if much more is allowed thing start to break

Reply:
think this is fine at least for now as before it was even more restrictive (e.g. using a `-` as first unquoted char also broke afaik)

Reply:
Looks like this caused a regression, previously `.` was allowed, now it isn't anymore